Framework/Muon/src/PSIBackgroundSubtraction.cpp

```diff
@@ -28,7 +28,7 @@ constexpr double FIT_TOLERANCE = 10;
 const std::string FIRST_GOOD = "First good spectra ";
 const std::string LAST_GOOD = "Last good spectra ";
 
-std::pair<double, double> getRange(const MatrixWorkspace_sptr &inputWorkspace, const size_t &index) {
+std::pair<double, double> getRangeFromWorkspace(MatrixWorkspace_const_sptr inputWorkspace, const size_t &index) {
```

Comment:
Just wondering why these shared_ptrs are now passed by value? Not 100% clear on shared_ptr semantics still, but my understanding is:

- Pass by value `(std::shared_ptr<A> a)` if you want to share ownership i.e increment the use_count
- Pass by reference `(std::shared_ptr<A>& a)` if you're not trying to share ownership (but may want to reset the pointer)
- Pass by const reference `(const std::shared_ptr<A>& a)`  if you don't want the function to be able to reset the shared ptr (which is what we had before) and don't extend the lifetime i.e. don't increment the use count

With the change, we are now passing by value `shared_ptr<const MatrixWorkspace>` meaning we are sharing ownership, i,e. incrementing the use count, but not letting the function modify the workspace. Is this the desired outcome? 

 I probably wrote the original function, so likely to blame here, but could getRangeFromWorkspace just take a bare reference e.g.

`getRangeFromWorkspace(const MatrixWorkspace& inputWorkspace, const size_t index)`

That way we don't have to care about the ownership debates that come with shared_ptrs.

Reply:
Thanks for taking a look, shared_ptr semantics are confusing!